pages/search.js

```diff
@@ -26,7 +26,27 @@ export async function getServerSideProps(context) {
     const res = await fetch(
       `${process.env.NEXT_PUBLIC_BASE_URL}/api/discover/tags`
     );
-    data.tags = await res.json();
+
+    const tags = await res.json();
+
+    const tagsWithOutDuplicates = {};
+
+    tags.forEach(({ name, total }) => {
+      const lowerName = name.toLowerCase();
+      if (tagsWithOutDuplicates[lowerName]) {
+        tagsWithOutDuplicates[lowerName] = {
+          ...tagsWithOutDuplicates[lowerName],
+          total: tagsWithOutDuplicates[lowerName].total + total,
+        };
+      } else {
+        tagsWithOutDuplicates[lowerName] = {
+          name: lowerName,
+          total,
+        };
+      }
+    });
```

Comment:
I think this could be done with a `reduce` that would not require the `if/else`

(not tested)
```js
const uniqueTags = tags.reduce((allTags, tag) => {
  const lowerTag = tag.toLowerCase();
  const currCount = allTags[lowerTag] ?? 0;
  return {
    ... allTags,
    [tag]: currCount + 1,
  };
}, {});
```

Reply:
Also, now we are getting the data from the database, we could use mongo's `group` and `count` feature in the `aggregation`

- https://www.mongodb.com/docs/manual/core/aggregation-pipeline/?_ga=2.243062658.1416307675.1684809031-436223288.1684493318
- https://www.mongodb.com/docs/manual/reference/operator/aggregation/group/?_ga=2.245751556.1416307675.1684809031-436223288.1684493318
- https://www.mongodb.com/docs/manual/reference/operator/aggregation/count/?_ga=2.245751556.1416307675.1684809031-436223288.1684493318

Reply:
Hey i dont know if i am supposed to write here or not 
sorry if i am not 

Have find a quick fix using aggregator as eddie mentioned 

i see aggregate is already implemented on [tags.js](https://github.com/EddieHubCommunity/LinkFree/blob/main/pages/api/discover/tags.js) file 

just if we convert all tags to lowercase it will filter all 

![image](https://github.com/EddieHubCommunity/LinkFree/assets/29349136/deb305dc-d3e4-47ed-a017-82a97647703b)

The fix i have found lemme know if this works or do i have to go more deep into this 
Thanks

Reply:
@shobhitexe thank you for the suggestion.  This does look like a possible solution (we have not tested this) but in addition when the search is being performed using the selected tag it would also need to be converted into lower case otherwise it would not return any results.

Reply:
Yes correct got it